### PR TITLE
[Passes] Resolving constant timing issue

### PR DESCRIPTION
**Pull request by yn224**
This PR attempts to address issue #486 by checking for a constant assignment to the hole port of the group before actually conducting static timing analysis on each group.

### CODE REVIEW

**Comment by rachitnigam** [comment]
@yn224 formatting errors. Run `cargo fmt` to reformat the code and use `cargo clippy` to fix common errors. `cargo clippy` might not report any errors if you don't run `cargo clean` first. Once done, please ping us for a review.

**Comment by rachitnigam** [comment]
Also, if you say "Fixes #486" in your issue description, github will automatically close that issue once this PR is merged.

**Comment by yn224** [comment]
@rachitnigam It seems like the `docs` testing is not passing but I'm not sure what the potential error would be.

File: tests/passes/guard-canonicalize.expect [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import "primitives/std.lib";
```
Comment (rachitnigam) [inline tests/passes/guard-canonicalize.expect:1]: I would have assumed that this file was added in #506 already?

**Comment by rachitnigam** [comment]
Looks good! Merge on green.

File: tests/passes/guard-canonicalize.expect (hunk 1 above)
Comment (yn224) [reply tests/passes/guard-canonicalize.expect]: Hmm.. I'm not sure why that did not get merged.

**Pull request merged**